include/Core/Config.php

```diff
@@ -132,15 +132,33 @@ public static function set($family,$key,$value) {
 		$dbvalue = (is_array($value)?serialize($value):$value);
 		$dbvalue = (is_bool($dbvalue) ? intval($dbvalue) : $dbvalue);
 
-		$ret = q("INSERT INTO `config` ( `cat`, `k`, `v` ) VALUES ( '%s', '%s', '%s' )
-ON DUPLICATE KEY UPDATE `v` = '%s'",
+		// The "INSERT" command is very cost intense. It saves performance to do it this way.
```

Comment:
The INSERT command is indeed cost intensive, but I feel like the best way to optimize saving settings would be not to call `PConfig::set()` in the first place.

The check you are doing here can be done much faster using the cached config values in the user/admin settings submit action, so that `PConfig::set()` would only be called for the values that changed. I'll submit a counter-PR to explain my point.

I disagree with this change.